src/jit/simd.cpp

```diff
@@ -1058,10 +1058,18 @@ GenTree* Compiler::impSIMDPopStack(var_types type, bool expectAddr, CORINFO_CLAS
     // If we have a ldobj of a SIMD local we need to transform it.
     if (tree->OperGet() == GT_OBJ)
     {
-        GenTree* addr = tree->gtOp.gtOp1;
-        if ((addr->OperGet() == GT_ADDR) && isSIMDTypeLocal(addr->gtOp.gtOp1))
+        if (tree->AsObj()->gtClass != structHandle)
```

Comment:
@CarolEidt  Shouldn't this check for `structHandle` being `NO_CLASS_HANDLE`? I think that's what's causing #23159

Reply:
I'll take a look at that (probably not until tomorrow), but I believe the issue would lie with the caller of `impSIMDPopStack` which should be supplying a valid class handle. If that's the case, I'll fix the caller and add an assert here. If there is indeed a case where the caller doesn't have a valid class handle then perhaps something else is needed.